### PR TITLE
Fix array index out of bounds in JFR Analyzer getAgentCaller method

### PR DESCRIPTION
**Pull request by hoteye**
## Summary
Fixed an `IndexOutOfBoundsException` in the `getAgentCaller` method of the JFR benchmark analyzer.

## Problem
The method had a bug where it could access `frames.get(i + 1)` when `i = frames.size() - 1` (first iteration of the loop), causing an array index out of bounds exception.

## Solution
- Changed loop condition from `i >= 0` to `i >= 1`
- Changed frame access from `frames.get(i + 1)` to `frames.get(i - 1)`

This maintains the correct logic to find the caller of agent methods while preventing array bounds violations.

## Test plan
- [x] Created and ran a test program that reproduces the bug and verifies the fix
- [x] Confirmed the original code throws `IndexOutOfBoundsException` with agent methods at stack top
- [x] Verified the fixed code handles all edge cases correctly (agent at top, middle, and no agent)

### CODE REVIEW

File: benchmark-jfr-analyzer/src/main/java/io/opentelemetry/javaagent/benchmark/jfr/Analyzer.java [hunk 1]
```diff
@@ -65,11 +65,11 @@ private static Set<String> getAgentCallers(List<RecordedEvent> events) {
   @Nullable
   private static String getAgentCaller(RecordedStackTrace stackTrace) {
     List<RecordedFrame> frames = stackTrace.getFrames();
-    for (int i = frames.size() - 1; i >= 0; i--) {
+    for (int i = frames.size() - 1; i >= 1; i--) {
```
Comment (trask) [inline benchmark-jfr-analyzer/src/main/java/io/opentelemetry/javaagent/benchmark/jfr/Analyzer.java]: hi @hoteye! sorry, I don't remember this code well, and we don't have any tests to verify, so could you include a bit more explanation? e.g., this seems like the simplest fix just for the index out of bounds issue (and not changing the access below)

```suggestion
    for (int i = frames.size() - 2; i >= 0; i--) {
```

File: benchmark-jfr-analyzer/src/main/java/io/opentelemetry/javaagent/benchmark/jfr/Analyzer.java [hunk 2]
```diff
@@ -65,11 +65,11 @@ private static Set<String> getAgentCallers(List<RecordedEvent> events) {
   @Nullable
   private static String getAgentCaller(RecordedStackTrace stackTrace) {
     List<RecordedFrame> frames = stackTrace.getFrames();
-    for (int i = frames.size() - 1; i >= 0; i--) {
+    for (int i = frames.size() - 2; i >= 0; i--) {
       RecordedFrame frame = frames.get(i);
       RecordedMethod method = frame.getMethod();
       if (isAgentMethod(method)) {
-        RecordedFrame callerFrame = frames.get(i + 1);
+        RecordedFrame callerFrame = frames.get(i - 1);
```
Comment (trask) [inline benchmark-jfr-analyzer/src/main/java/io/opentelemetry/javaagent/benchmark/jfr/Analyzer.java]: is this change needed now that you've updated the bounds above?

**Review by trask — APPROVED** [review]
